### ISSUE TITLE
Eclipse complains about Java version when using Gradle and Java 9

### ISSUE THREAD

**Issue by donat** [eclipse/buildship#598]
[Report](https://bugs.eclipse.org/bugs/show_bug.cgi?id=526265#add_comment) from Umar Osman:

> When using Java 9 and Gradle, after refreshing the Gradle project errors start popping up on a lot of standard Java objects and such (e.g annotations, HashMaps) saying that it is only implemented on Java 1.5 or higher. This was tested on 2 different computers, and neither of them had ever had a Java version lower that 1.6 on them.

**Comment by cobexer** [comment]
In a project with `sourceCompatibility = 9` and `targetCompatibility = 9` imported into Eclipse I get the above mentioned compile errors. Looking into the generated settings file I see:

```$ cat .settings/org.eclipse.jdt.core.prefs 
eclipse.preferences.version=1
org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.9
org.eclipse.jdt.core.compiler.compliance=1.9
org.eclipse.jdt.core.compiler.source=1.9
```

If I change all the `1.9` to just `9` and then clean & rebuild in Eclipse the errors go away.

Considering that the execution environment was fixed in https://github.com/gradle/gradle/issues/3046 should this issue actually move there?

**Comment by donat** [comment]
@cobexer I've just pushed the fix for this problem in the #608 pull request. It will take some time until we finish the review and publish the changes in a snapshot/final release. In case you are impatient you can install a development build from [this update site](https://builds.gradle.org/repository/download/Tooling_Master_IntegrationTests_Linux_Eclipse47IntegrationTests/9039151:id) (if prompted, specify guest/guest for the credentials). It contains everything from the latest snapshot release plus the fixes for Java 9.

**Comment by cobexer** [comment]
Ah that's great to hear! I'm not in a hurry with Java 9 and can manually patch projects locally for testing... no big deal. Thanks for the quick reply!

**Comment by donat** [comment]
Fixed in 6ff9af2813416603a4f72cc4131c797092ddc497
